internal/rancher/provisioning/v1/cluster.go

```diff
@@ -0,0 +1,34 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Cluster is the struct representing a Rancher Cluster.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Status ClusterStatus `json:"status,omitempty"`
```

Comment:
nit: we can add comments to the fields as well. But its not needed for this and we can follow up on it.